src/coreclr/jit/morph.cpp

```diff
@@ -5465,11 +5465,21 @@ GenTree* Compiler::fgMorphArrayIndex(GenTree* tree)
 
     GenTree* cns = gtNewIconNode(elemOffs, TYP_I_IMPL);
 
+// We're going to create a platform-specific address:
```

Comment:
You should add a note to the comment above (starting "// Be careful to only create the byref") explaining why this change here is ok.